src/move_basic.cpp

```diff
@@ -523,20 +547,55 @@ bool MoveBasic::moveLinear(double requestedDistance)
          return false;
     }
 
+    double requestedDistance = (poseOdomInitial.getOrigin() -
+                                goalInOdom.getOrigin()).length();
+
+    // For lateral control
+    double lateralIntegral = 0.0;
+    double lateralPrevError = 0.0;
+    double lateralError = 0.0;
+
     while (!done && ros::ok()) {
         ros::spinOnce();
         r.sleep();
 
         tf2::Transform poseOdom;
-        if (!getTransform(baseFrame, odomFrame, poseOdom)) {
+        if (!getTransform(odomFrame, baseFrame, poseOdom)) {
              ROS_WARN("Cannot determine robot pose for linear");
              continue;
         }
 
-        tf2::Vector3 travelled = poseOdomInitial.getOrigin() - poseOdom.getOrigin();
-        double distTravelled = travelled.length();;
+        tf2::Transform goalInBase = poseOdom * goalInOdom;
+        tf2::Vector3 remaining = goalInBase.getOrigin();
+
+        double distRemaining = remaining.x();
+        double distTravelled = std::abs(requestedDistance) - std::abs(distRemaining);
+
+        // PID loop to control rotation to keep robot on path
+        double rotation = 0.0;
+
+        lateralPrevError = lateralError;
+        lateralError = remaining.y();
+        double lateralDiff = lateralError - lateralPrevError;
+        lateralIntegral += lateralError;
+
+        rotation = (lateralKp * lateralError) + (lateralKi * lateralIntegral) +
+                   (lateralKd * lateralDiff);
+
+        // Limit rotation
+        if (rotation > lateralMaxRotation) {
```

Comment:
std::min/std::max for clamping?

Reply:
Yes, that's what we used in other places in the file, so I'll change it.